### PR TITLE
XW-2307 | in infobox only image files are displayed

### PR DESCRIPTION
**Pull request by weronika**
https://wikia-inc.atlassian.net/browse/XW-2307

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [PortableInfoboxRenderServiceHelper.php#L71](https://github.com/Wikia/app/blob/acaa9fb77a48274bb2930cebb6bbd3c63ab26ce2/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L71): Define a constant instead of duplicating this literal "width" 8 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L75](https://github.com/Wikia/app/blob/acaa9fb77a48274bb2930cebb6bbd3c63ab26ce2/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L75): Define a constant instead of duplicating this literal "height" 7 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L95](https://github.com/Wikia/app/blob/acaa9fb77a48274bb2930cebb6bbd3c63ab26ce2/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L95): Define a constant instead of duplicating this literal "image" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L114](https://github.com/Wikia/app/blob/acaa9fb77a48274bb2930cebb6bbd3c63ab26ce2/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L114): Immediately return this expression instead of assigning it to the temporary variable "$data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [PortableInfoboxRenderServiceHelper.php#L71](https://github.com/Wikia/app/blob/b21dc5101aa178dc55fba38dfc1acfa55c485039/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L71): Define a constant instead of duplicating this literal "width" 8 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L75](https://github.com/Wikia/app/blob/b21dc5101aa178dc55fba38dfc1acfa55c485039/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L75): Define a constant instead of duplicating this literal "height" 7 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L95](https://github.com/Wikia/app/blob/b21dc5101aa178dc55fba38dfc1acfa55c485039/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L95): Define a constant instead of duplicating this literal "image" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L114](https://github.com/Wikia/app/blob/b21dc5101aa178dc55fba38dfc1acfa55c485039/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L114): Immediately return this expression instead of assigning it to the temporary variable "$data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

File: extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php [hunk 1]
```diff
@@ -57,9 +57,11 @@ public function extendImageData( $data, $width ) {
 		// title param is provided through reference in WikiaFileHelper::getFileFromTitle
 		$title = $data[ 'name' ];
 		$file = \WikiaFileHelper::getFileFromTitle( $title );
-		if ( !$file || !$file->exists() ) {
+
+		if ( !$file || !$file->exists() || !in_array($file->media_type, array(MEDIATYPE_BITMAP, MEDIATYPE_DRAWING))) {
```
Comment (Warkot) [inline extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php]: https://www.mediawiki.org/wiki/Manual:Coding_conventions/PHP#Spaces 🚀

**Comment by Warkot** [comment]
Beside one minor comment (spaces), LGTM. 👍

**Review by Warkot — APPROVED** [review]

File: extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php (hunk 1 above)
Comment (kvas-damian) [reply extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php]: What do you think about using `$file->getMediaType()`? 
https://github.com/Wikia/app/blob/dev/includes/filerepo/file/LocalFile.php#L584

Comment (kvas-damian) [reply extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php]: new array syntax ;)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 39 issues
* MAJOR 1 major
* MINOR 38 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PortableInfoboxRenderServiceHelperTest.php#L40](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L40): Remove this unused private "getThumbnailMock" method. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1144)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L71](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L71): Define a constant instead of duplicating this literal "width" 8 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L75](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L75): Define a constant instead of duplicating this literal "height" 7 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L95](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L95): Define a constant instead of duplicating this literal "image" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L114](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L114): Immediately return this expression instead of assigning it to the temporary variable "$data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L21](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L21): Define a constant instead of duplicating this literal "fileWidth" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L26](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L26): Define a constant instead of duplicating this literal "getWidth" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L26](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L26): Define a constant instead of duplicating this literal "getHeight" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L41](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L41): Define a constant instead of duplicating this literal "width" 35 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L42](https://github.com/Wikia/app/blob/123a5b9ef59df6841d309b18ffb60c6c062b581b/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L42): Define a constant instead of duplicating this literal "height" 35 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Review by kvas-damian — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 39 issues
* MAJOR 1 major
* MINOR 38 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PortableInfoboxRenderServiceHelperTest.php#L40](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L40): Remove this unused private "getThumbnailMock" method. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1144)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L72](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L72): Define a constant instead of duplicating this literal "width" 8 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L76](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L76): Define a constant instead of duplicating this literal "height" 7 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L96](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L96): Define a constant instead of duplicating this literal "image" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelper.php#L115](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php#L115): Immediately return this expression instead of assigning it to the temporary variable "$data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L21](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L21): Define a constant instead of duplicating this literal "fileWidth" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L26](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L26): Define a constant instead of duplicating this literal "getWidth" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L26](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L26): Define a constant instead of duplicating this literal "getHeight" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L41](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L41): Define a constant instead of duplicating this literal "width" 35 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PortableInfoboxRenderServiceHelperTest.php#L42](https://github.com/Wikia/app/blob/12a4c6207113de9584cdf8104cb9329b62790eab/extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php#L42): Define a constant instead of duplicating this literal "height" 35 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Pull request merged**